Dump Stack when docker fails on healthcheck

tomoe opened the pull request:
Save stack dump of docker daemon in order to be able to
investigate why docker daemon was unresposive to `docker ps`

See https://github.com/moby/moby/blob/master/daemon/daemon.go on
how docker sets up a trap for SIGUSR1 with `setupDumpStackTrap()`

**What this PR does / why we need it**:

This allows us to investigate why docker daemon was unresponsive to "docker ps" command. 

**Special notes for your reviewer**:
Manually tested on Ubuntu and COS.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

wgliang commented:
/ok-to-test

krmayankk commented on cluster/gce/gci/health-monitor.sh:
```diff
@@ -28,6 +28,11 @@ function docker_monitoring {
+      # Dump stack of docker daemon for investigation.
+      # Log fle name looks like goroutine-stacks-TIMESTAMP and will be saved to
+      # the exec root directory, which is /var/run/docker/ on Ubuntu and COS.
+      pkill -SIGUSR1 docker
```
in order to take the dump, i send SIGUSR1 to dockerd or docker-containerd process. Not sure what `docker` would do  here. Would this also send the signal to the daemon `dockerd` ?

tomoe replied:
Thanks @krmayankk. Naively, I tried to be consistent with the following line, but just realized that the pattern `docker` would match against `dockerd`, `docker-containerd`, and `docker-containerd-shim`. Will change to `dockerd`

tomoe replied:
Changed to `dockerd`

tomoe committed:
Dump Stack when docker fails on healthcheck

Send SIGUSR1 to dockerd to save stack dump of docker daemon
in order to be able to investigate why docker daemon was
unresposive to health check done by `docker ps`.

See https://github.com/moby/moby/blob/master/daemon/daemon.go on
how docker sets up a trap for SIGUSR1 with `setupDumpStackTrap()`

tomoe commented:
/test pull-kubernetes-e2e-gke

tomoe commented:
/test pull-kubernetes-verify

tomoe commented:
Hi @jingax10 jingax10 Could you take a look?

jingax10 commented:
/lgtm
/approve no-issue
